tests/system/src/test/java/test/robot/javafx/web/PointerEventTest.java

```diff
@@ -0,0 +1,229 @@
+            stage.show();
+        }
+    }
+
+    public String mouseButtonDrag(MouseButton... button) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int)(scene.getWindow().getX() + scene.getX() + DX),
+                        (int)(scene.getWindow().getY() + scene.getY() + DY ));
+            robot.mousePress(button);
+        });
+
+        for (int i = 0; i < DRAG_DISTANCE; i++) {
+                final int c = i;
+                Util.runAndWait(() -> {
+                    // Move the mouse backwards so that the pointer does not stay on the popup,if any.
+                    robot.mouseMove((int)(scene.getWindow().getX() + scene.getX() + DX - c),
+                        (int)(scene.getWindow().getY() + scene.getY() + DY) );
+            });
+        }
+
+        Util.sleep(SLEEP_TIME);
+        Util.runAndWait(()  -> {
+            buttonMask = element.getTextContent();
+            robot.mouseRelease(button);
+        });
+
+        return buttonMask;
+    }
+
+    @Test
+    public void testLeftButtonDrag() {
+        int result = Integer.parseInt(mouseButtonDrag(MouseButton.PRIMARY));
+        Assert.assertEquals(LEFT_BUTTON_DRAG,result);
+    }
+
+    @Test
+    public void testRightButtonDrag() {
+        int result = Integer.parseInt(mouseButtonDrag(MouseButton.SECONDARY));
+        Assert.assertEquals(RIGHT_BUTTON_DRAG,result);
+    }
+
+    @Test
+    public void testMiddleButtonDrag() {
+        int result = Integer.parseInt(mouseButtonDrag(MouseButton.MIDDLE));
+        Assert.assertEquals(MIDDLE_BUTTON_DRAG,result);
+    }
+
+    @Test
+    public void testLeftMiddleButtonDrag() {
+        int result = Integer.parseInt(mouseButtonDrag(MouseButton.PRIMARY, MouseButton.MIDDLE));
+        Assert.assertEquals((LEFT_BUTTON_DRAG|MIDDLE_BUTTON_DRAG),result);
+    }
+
+    @Test
+    public void testMiddleRightButtonDrag() {
+        int result = Integer.parseInt(mouseButtonDrag(MouseButton.MIDDLE,MouseButton.SECONDARY));
+        Assert.assertEquals((MIDDLE_BUTTON_DRAG|RIGHT_BUTTON_DRAG),result);
+    }
+
+    @Test
+    public void testLeftRightButtonDrag() {
+        int result = Integer.parseInt(mouseButtonDrag(MouseButton.PRIMARY, MouseButton.SECONDARY));
+        Assert.assertEquals((LEFT_BUTTON_DRAG|RIGHT_BUTTON_DRAG),result);
+    }
+
+    @Test
+    public void testLeftMiddleRightButtonDrag() {
+        int result = Integer.parseInt(mouseButtonDrag(MouseButton.PRIMARY, MouseButton.MIDDLE, MouseButton.SECONDARY));
+        Assert.assertEquals((LEFT_BUTTON_DRAG|MIDDLE_BUTTON_DRAG|RIGHT_BUTTON_DRAG),result);
+    }
+
+    @BeforeClass
+    public static void initFX() {
+        startupLatch = new CountDownLatch(1);
+        loadLatch  = new CountDownLatch(1);
+        new Thread(() -> Application.launch(TestApp.class, (String[])null)).start();
+        waitForLatch(loadLatch, 10, "Timeout waiting for loading webpage().");
+        waitForLatch(startupLatch, 15, "Timeout waiting for FX runtime to start");
```

Comment:
Since `startupLatch` will trigger first, you should either switch the two calls, or use a single latch (initialized to 2) with a single call to `waitForLatch`.

Reply:
Added single latch initialised to 2.